x-pack/plugins/task_manager/server/task.ts

```diff
@@ -88,6 +89,8 @@ export type FailedRunResult = SuccessfulRunResult & {
 
 export type RunResult = FailedRunResult | SuccessfulRunResult;
 
+export const DELETE_TASK_RUN_RESULT: SuccessfulRunResult = { state: {}, shouldDeleteTask: true };
```

Comment:
just happened to think.  What if someone changes some of the properties in this object, accidentally.  Should be legal, but obviously not something we want to happen.  We could make this a function which returns a copy of the object each time, or we could try to ["freeze"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) it or something?  I've never played with `freeze()` or friends, not sure if there are any "issues" with using it ...

Reply:
Good call! Updated to a function in https://github.com/elastic/kibana/pull/184776/commits/00221b0d335b7fffc0c894d399d9e24988a66fae